concordium-consensus/src/Concordium/Scheduler.hs

```diff
@@ -2656,6 +2664,49 @@ handleUpdateCredentialKeys wtc cid keys sigs =
         updateCredentialKeys (fst senderAccount) index keys
         return (TxSuccess [CredentialKeysUpdated cid], energyCost, usedEnergy)
 
+-- | Handler for a token holder transaction.
+handleTokenHolder ::
+    ( PVSupportsPLT (MPV m),
+      SchedulerMonad m
+    ) =>
+    WithDepositContext m ->
+    -- | Token symbol identifying the token to receive the operations.
+    TokenId ->
+    -- | Operations for the token.
+    TokenParameter ->
+    m (Maybe TransactionSummary)
+handleTokenHolder depositContext tokenId tokenOperations =
+    withDeposit depositContext computeTransaction commitTransaction
+  where
+    senderAccount = depositContext ^. wtcSenderAccount
+    -- Execute the transaction in a modified environment
+    computeTransaction = do
+        -- Charge the base cost for this transaction type
+        tickEnergy Cost.tokenHolderBaseCost
+        -- Fetch the token from state
+        tokenIndex <-
+            lift (getTokenIndex tokenId) >>= \case
+                Just tokenIndex -> return tokenIndex
+                Nothing -> rejectTransaction $ NonExistentTokenId tokenId
+        -- Fetch the token configuration to find the reference for the token module.
+        configuration <- lift $ getTokenConfiguration tokenIndex
+        let moduleRef = Token._pltModule configuration
```

Comment:
I'm not sure we should do this here. At the moment, it's not used, and I think it could be handled further down the stack when it eventually is.

Reply:
That is also an option. Don't really have an opinion on that, so feel free to change it as we introduce the token module.

Reply:
I mainly wanted to prepare for charging the cost of loading the module, but that mights as well be further down the stack as well